File: my_code.java
import java.util.*;
class Student {
    String studentId;
    String name;
    int age;
    String grade;

    public Student(String studentId , String name, int age,String grade){
        this.studentId = studentId;
        this.name = name;
        this.age = age;
        this.grade = grade;
    }
    
}
public class my_code {
    public static void add_student(ArrayList<Student> list , Scanner sc){
        System.out.print("Enter Student ID: ");
        String id = sc.nextLine();
        System.out.print("ENTER STUDENT NAME: ");
        String name = sc.nextLine();
        System.out.print("ENTER STUDENT AGE: ");
        int age = sc.nextInt();
        sc.nextLine();
        System.out.print("ENTER STUDENT GRADE: ");
        String grade = sc.nextLine();

        Student student = new Student(id, name, age, grade);
        list.add(student);
        System.out.println("-------- STUDENT ADDED SUCCESFULLY -------");
    }
    public static void search_student(ArrayList<Student> list , Scanner sc){
        System.out.println("ENTER STUDENT ID TO SEARCH: ");
        String id = sc.nextLine();

        for( Student student : list){
            if( student.studentId.equals(id)){
                System.out.println("-------------------------------------");
                System.out.println("STUDENT FOUND:\nID: " + student.studentId + "\nName: " + student.name +
                        "\nAge: " + student.age + "\nGrade: " + student.grade);
                System.out.println("-------------------------------------");
                return;
            }
        }
        System.out.println("ID  DOES NOT EXIST");
    }
    public static void remove_student(ArrayList<Student> list , Scanner sc){
        System.out.println("ENTER STUDENT ID TO REMOVE : ");
        String id = sc.nextLine();

        for( Student student : list){
            if( student.studentId.equals(id)){
                list.remove(student);
                System.out.println("---------- STUDENT REMOVED SUCCESFULLY ---------");
                return;
            }
        }
        System.out.println("ID  DOES NOT EXIST");
    }
    public static void show_student(ArrayList<Student> list){
        if(list.isEmpty()){
            System.out.println("----------- NO STUDENTS IN RECORD ----------");
            return;
        }
        for( Student student : list){
                System.out.println("-------------------------------------");
                System.out.println("ID: " + student.studentId + "\nName: " + student.name +
                        "\nAge: " + student.age + "\nGrade: " + student.grade);
                System.out.println("-------------------------------------");
                
        }
        return;
    }
    public static void update_student(ArrayList<Student> list, Scanner sc){
        System.out.println("ENTER STUDENT ID TO UPDATE : ");
        String id = sc.nextLine();
       
        System.out.println("SELECT THE FEILD TO UPDATE\n1. ID\n2. NAME\n3. AGE\n4. GRADE\n5. EXIT");
        int choice = sc.nextInt();
        sc.nextLine();

        for(Student student : list){
            if(student.studentId.equals(id)){
                switch (choice) {
                    case 1:
                        System.out.print("ENTER ID TO UPDATE : ");
                        String new_id = sc.nextLine();
                        student.studentId = new_id;
                        break;
                    case 2:
                        System.out.print("ENTER NAME TO UPDATE : ");
                        String new_name = sc.nextLine();
                        student.name = new_name;
                        break;
                    case 3:
                        System.out.print("ENTER AGE TO UPDATE : ");
                        int new_age = sc.nextInt();
                        sc.nextLine();
                        student.age = new_age;
                        break;
                    case 4:
                        System.out.print("ENTER GRADE TO UPDATE : ");
                        String new_grade = sc.nextLine();
                        student.grade = new_grade;
                        break;
                    case 5:
                        System.out.println("---------- STUDENT DETAILS UPDATED SUCCESFULLY ---------");
                        return;
                    default:
                        System.out.println("Invalid choice. Please enter a number between 1 and 5.");
                        break;
                }
            }
        }
    }
    public static void main(String[] args){
        ArrayList<Student> list = new ArrayList<>();
        Scanner sc = new Scanner(System.in);

        while (true) {
            System.out.println("MENU\n1. ADD STUDENT\n2. SEARCH STUDENT\n3. REMOVE STUDENT\n4. SHOW STUDENT\n5. UPDATE STUDENT DEATILS\n6. EXIT");
            System.out.print("ENTER YOUR CHOICE: ");
            int choice = sc.nextInt();
            sc.nextLine();
            switch (choice) {
                case 1:
                    add_student(list,sc);
                    break;
                case 2:
                    search_student(list,sc);
                    break;
                case 3:
                    remove_student(list,sc);
                    break;
                case 4:
                    show_student(list);
                    break;
                case 5:
                    update_student(list,sc);
                    break;
                case 6:
                    System.out.println("--------- END ---------");
                    sc.close();
                    System.exit(0);
            
                default:
                    System.out.println("Invalid choice. Please enter a number between 1 and 5.");
            }
        }

    }
}

File: my_code.py
class Student:
    def __init__(self, id, name , age, grade):
        self.id = id
        self.name = name
        self.age = age
        self.grade = grade

def add_student(list):
    print("Enter Student ID : ")
    id = input()
    print("ENTER STUDENT NAME : ")
    name = input()
    print("ENTER STUDENT AGE : ")
    age = int(input())
    print("ENTER STUDENT GRADE : ")
    grade = input()
    
    new_student = Student(id,name,age,grade)
    list.append(new_student)
    print("-------- STUDENT ADDED SUCCESFULLY -------")

def search_student(list):
    print("ENTER STUDENT ID TO SEARCH : ")
    id = input()
    
    for student in list:
        if student.id == id:
            print("-------------------------------------")
            print(f"STUDENT FOUND:\nID: {student.id}\nName: {student.name}\nAge: {student.age}\nGrade: {student.grade}")
            print("-------------------------------------")
            return 
    
    print("ID  DOES NOT EXIST")
    
def remove_student(list):
    print("ENTER STUDENT ID TO REMOVE : ")
    id = input()
    
    for student in list:
        if student.id == id:
            list.remove(student)
            print("---------- STUDENT REMOVED SUCCESFULLY ---------")
            return 
        
    print("ID  DOES NOT EXIST")
    
def show_student(list):
    
    if not list:
        print("----------- NO STUDENTS IN RECORD ----------")
    
    for student in list:
         print("-------------------------------------")
         print(f"STUDENT FOUND:\nID: {student.id}\nName: {student.name}\nAge: {student.age}\nGrade: {student.grade}")
         print("-------------------------------------")

def update_student(list):
    print("ENTER STUDENT ID TO UPDATE : ")
    id = input()
    
    print("SELECT THE FEILD TO UPDATE\n1. ID\n2. NAME\n3. AGE\n4. GRADE\n5. EXIT")
    choice = int(input())
    
    for student in list:
        if student.id == id:
            if choice == 1:
                print("ENTER ID TO UPDATE : ")
                id = input()
                student.id = id
            elif choice == 2:
                new_name = input("Enter NAME to update : ")
                student.name = new_name
            elif choice == 3:
                new_age = int(input("Enter AGE to update : "))
                student.age = new_age
            elif choice == 4:
                new_grade = input("Enter GRADE to update : ")
                student.grade = new_grade
            elif choice == 5:
                print("---------- STUDENT DETAILS UPDATED SUCCESSFULLY ---------")
                return
            else:
                print("Invalid choice. Please enter a number between 1 and 5.")
                return
students_list = []

while True:
    print("MENU\n1. ADD STUDENT\n2. SEARCH STUDENT\n3. REMOVE STUDENT\n4. SHOW STUDENT\n5. UPDATE STUDENT DEATILS\n6. EXIT")
    choice = int(input("ENTER YOUR CHOICE: "))

    if choice == 1:
        add_student(students_list)
    elif choice == 2:
        search_student(students_list)
    elif choice == 3:
        remove_student(students_list)
    elif choice == 4:
        show_student(students_list)
    elif choice == 5:
        update_student(students_list)
    elif choice == 6:
        print("--------- END ---------")
        break
    else:
        print("Invalid choice. Please enter a number between 1 and 6.")
                
            
    